src/Persistence/FromPageStatementsLookup.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\WikibaseFacetedSearch\Persistence;
+
+use ProfessionalWiki\WikibaseFacetedSearch\Application\StatementsLookup;
+use Wikibase\DataModel\Statement\StatementList;
+use Wikibase\Repo\Content\ItemContent;
+use WikiPage;
+
+class FromPageStatementsLookup implements StatementsLookup {
+
+	public function getStatements( WikiPage $page ): StatementList {
+		$content = $page->getContent();
+
+		if ( !( $content instanceof ItemContent ) ) {
+			return new StatementList();
+		}
+
+		return $content->getItem()->getStatements();
```

Comment:
After some consideration, I think this code is great.

There is also opportunity to check for `EntityContent` and then see if `getEntity` returns `StatementListProvider`, so we support non-Items. Unclear why we would need that. Too long, don't read: https://www.entropywins.wtf/blog/2017/05/24/generic-entity-handling-code/

Reply:
I tried using Item-specific implementations where they were available. 

While it is indeed technically possible to handle non-Items, my main reasons against doing that:
* Property cannot have a sitelink. This is not technically relevant in this class, but leads to a subset of features compared to Item. Additionally, if the sitelink is configured, none of the Properties will get indexed, since there won't be a linked Page.
* Property is unlikely to have an "instance type" statement. (Property Foo might be an instance of "XYZ Property", but not something like "Person"). Of course, technically it is possible.

If not for above use case, but rather as a more general entity statement lookup - I also don't see a need for that at the moment.